### ISSUE TITLE
Docs / wiki for new items added that aren't on curse (aka how to use item pickup manager)

### ISSUE THREAD

**Issue by DoomSquirter** [maruohon/enderutilities#28]
Could you point me towards any documentation or anything that shows how to use some of the newer items added that aren't already on curse?

For example, the item pickup manager.  Thought this might be a workaround for not having ability to bind to a specific chest with the ender bag but for the life of me, I can't figure out how it works.  I added a crystal that was bound to a block.  added a capacitor and it says enabled but nothing I pickup gets put into that inventory.  

Thanks for making probably my favorite mod of 1.10 thus far.  My favorite items are the Handy Bag (which I think is the best item -period- in modded minecraft), and the handy chests and the lazy builders wand and especially the elevators.   You've definately found a niche that needed addressing.  

Edit: Ok, I think I figured out what I was doing wrong, I assume for the item pickup manager you need to use filters.  there is no "pickup everything not in hotbar" kind of thing.  I can deal with that :)  I guess the only question I still have is if there are plans to update the curse page with more info or if you're setting up a wiki or something.  Thanks again!

**Comment by maruohon** [comment]
Yes unfortunately I haven't updated the CurseForge description for anything added in the 0.5.x or later versions and I'm sorry for that :/ All my other mods have up-to-date documentation except Ender Utilities for the newer and more useful content. I have meant to start writing it many times, but then something else has distracted me, since writing documentation usually isn't the most interesting of things :P Things like the Wand of the Lazy Builder and the Item Pickup Manager definitely would benefit from proper documentation. I'll try to work on that "soon" (tm).

About the I.P.M. on particular: the main (ie. "pickup") functionality of it is only triggered when the player would pick an item from the ground (which in code happens when the player entity collides with an item entity in world, more or less). So the idea of it is that you can do two separate things with the items that you would normally pick up to your inventory:

1) You can configure filters (black or white list) for automatically sending the would-be-picked-up items to the linked target inventory (and sending the items is the only thing that needs the capacitor/ender charge in this item btw). You can also manually send items by putting them/shift clicking them into the transport slot (poor man's ender pouch, kind of).

2) You can either black list or white list a set of items that you are allowed to pick up from the ground to your inventory.

For these tasks there are the two separate filter sections. The remote transport is first, if an item is allowed through that filter, then it will immediately be transported (if there is space in the target inventory). If an item can't or isn't allowed to be transported, or if the transport feature is bypassed/disabled (the arrow buttons on the side), then the inventory pickup filter is considered next. Same deal here, you can bypass the filter, which means that you would pick up everything as normal. If it's enabled, then you can ONLY pick up items that pass the filter settings. So let's say you have it as black list, with dirt and cobble as the filter items. Then you would not be able to pick up dirt or cobble items from the ground as long as the I.P.M. is in your inventory, and enabled.

Also note that there is support for multiple I.P.M. items. The way that works is that the IPMs are looked at in order in your inventory, and the first one that has a "direct match" to the item in question is the one that defines what happens to it. This means that when the item is actually set as a filter, either black or white list, then that is what happens to that item (from the first IPM found, like I said above). So if you have a white list for diamonds, and you try to pick up dirt, since it's not directly found (disallowed, but not black listed per se), it will look at the subsequent IPMs for matches. If there is another IPM with white list dirt, then the dirt will be allowed. If there are no matches, then it will NOT be allowed since the first one had a white list and the item was not allowed based on that. The order in which IPMs are looked at is top left to bottom right, but hotbar is the first row (it's defined that way in code, only visually in the GUI the hotbar is the lowest row).

So to sum it up a bit, you can configure an IPM to send random crap like common mob drops or plant stuff etc. directly to storage, or a trash can or whatever, and/or you can configure which items you are allowed to pick up from the ground while running around. Or you can use the IPM just for manually sending items to the linked target inventory.

**Comment by DoomSquirter** [comment]
jeez that is so damn useful lol.  first wall of text I totally approved :)  

I will say this, going through jei on your items and holding shift does give you some sort of awe / fear when you see all the options presented to you.  I'm really on the fence whether or not having all that info is perfect for the complex objects you are presenting or if it'd be more serving for a in game doc.  TBH I don't know.  

As a reference, from the little info I could find on the wand of the lazy builder for example, it took me about 6+ hours to copy my first area.  The first (6 hrs - 10 minutes) was trying to figure out how to mark things with it.  Once I had that it was smooth sailing.  

I'm a 3x3 chunk base builder on the server I play on alot and ppl tend to make fun of me for it.  Using that thing, I was able to take my hollowed out base down to bedrock via rftools builder (another fantastic mod, now my second favorite) and make all my levels with lights, etc.  gave me much satisfaction.  

Thanks for the info.  I'll close this issue and I apologize for wasting ur time :)

**Comment by maruohon** [comment]
Yep the Wand of the Lazy Builder is a rather complex item and I'm sure it will be scary and daunting for new players. I have tried to make all they bazillions of keybinds and other functions it has as consistent as possible though. So for example all modes that have two corners use left click to place the "primary" corner and right click to place the other, "secondary" corner. If the mode doesn't use bound blocktypes, then sneaking will set the position inside the block instead of against it. If the mode uses bound blocktypes, then sneak + left clicking binds a block to the currently selected slot/index/whatever. When the primary (or only) position has been set, then shift + scrolling and looking at the direction you want to change the size in, will change the area size or move the secondary position.

I will also try to make a dedicated spotlight of it after I hopefully some day finish it :P Luckily there aren't that many more things planned for it anymore. My dev version already has a completed Stack mode, and the next things I will be adding are Circle and Sphere modes. And after that I want it to also have Undo functionality, and then hopefully I'll call it done.

It would also be nice to have an in-game manual for some of my more complex items, the wand definitely being one of those. Hopefully I'll get to that at some point... >_>

I have to say... if it took 6 hours to learn how to use the wand (in one mode), then I have some work to do with the documentation of it, haha :D
